internal/plugins/plugins.go

```diff
@@ -146,11 +155,86 @@ func createCmdRunE(pluginExecutable string) func(cmd *cobra.Command, args []stri
 			return err
 		}
 
-		err = plugin.Run(args, &shared_logger.SharedLogger{})
+		err = plugin.Run(pluginArgs, &shared_logger.SharedLogger{})
 		if err != nil {
 			return fmt.Errorf("failed to execute plugin command: %w", err)
 		}
 
 		return nil
 	}
 }
+
+// filterRootFlags filters out any flags that were parsed by the root command's persistent flags
+// and processes them for the host application, returning only plugin-specific args.
+func filterRootFlags(args []string, persistentFlags *pflag.FlagSet) ([]string, error) {
+	pluginArgs := []string{}
+
+	var (
+		previousArgFlagName     = ""
+		handlePreviousArgAsFlag = false
+	)
+
+	for _, arg := range args {
+		switch {
+		case handlePreviousArgAsFlag && previousArgFlagName != "":
+			err := persistentFlags.Set(previousArgFlagName, arg)
+			if err != nil {
+				return nil, fmt.Errorf("failed to set persistent flag '%s' with value '%s': %w", previousArgFlagName, arg, err)
+			}
+			handlePreviousArgAsFlag = false
+		case len(arg) > 0 && arg[0] == '-':
+			// The argument is a flag, remove leading dashes
+			flagArg := strings.TrimLeft(arg, "-")
+
+			// Handle flags in the format --flag=value
+			if strings.Contains(flagArg, "=") {
+				parts := strings.SplitN(flagArg, "=", 2)
+				flagName := parts[0]
+
+				if flag := persistentFlags.Lookup(flagName); flag != nil {
+					err := persistentFlags.Set(flagName, parts[1])
```

Comment:
Why are these blocks with `Set` needed? I would have thought that as soon as the Lookup finds something we can just move on.

Reply:
Set is needed because we turned off flag parsing for the host process, and so we need to set the flag's value if it is a persistent (global) flag.